main/ControllersPresenters/InterviewLevel.java

```diff
@@ -34,22 +33,63 @@ public String getInterviewOutput(String input) throws Exception{
             return getEndOfInterviewPrompt();
         }
         if (currentInterviewMaker.getChoiceOptions().contains(input)) {
-            //check if this is the last thing the intern says
-            //if it's not:
-                // return the corresponding interviewee's next response and next choice options to the player
-            //if it is:
-                //return the interviewee's final response, and prompt player to make hiring decision.
-                //this.getHiringResponse(input);
+            //if the current InterviewIntern has not yet said their last response:
+            if (! this.currentInterviewMaker.checkInternsLastResponse(input)){
+                return GamePrompts.INTERN_RESPONSE_PROMPT +
+                        currentInterviewMaker.displayInternChoiceResponse(input) +
+                        GamePrompts.NEXT_CHOICE_PROMPT +
+                        currentInterviewMaker.getChoiceOptions();
+            }
+
+            //if the current InterviewIntern has said their last response (i.e. at the end of the interview):
+            if (this.currentInterviewMaker.checkInternsLastResponse(input)){
+                return GamePrompts.HIRE_INTERN + this.currentInterviewMaker.internToHire(input);
+            }
+
         }
         if (Objects.equals(input, "yes") || (Objects.equals(input, "no"))){
             //hired the intern/ or not
             //return "successfully/ don't hired intern"
-            //check if there are any interns left to interview haveInterviewsLeft()
-            // if yes: update the current interviewing intern, then also add the next interviewee's next sentenct to output
-            // if no: endLevel() and return hired successful or not
-            currentInterviewMaker.hireIntern();//the actual hiring
+            if (Objects.equals(input, "yes")){
+                this.currentInterviewMaker.hireIntern();
+                //check if there are any interns left to interview haveInterviewsLeft()
+                //if there are more interns to interview, update the current interviewing intern, prompt the player from the
+                // next interviewee's output
+                if (this.currentInterviewMaker.haveInterviewsLeft()){
+                    this.currentInterviewMaker.updateInterviewIntern();
+                    return GamePrompts.HIRED_INTERN + GamePrompts.NEXT_INTERVIEW_INTERN_PROMPT +
+                            this.currentInterviewMaker.getInterviewInternInfo();
```

Comment:
does method getInterviewInternInfo include the choices that the player can make?

Reply:
if not, then it doesn't return everything we need

Reply:
yes, the second if condition in getInterviewInternInfo should return the first choices to pick from

Reply:
Ok, see method getInterviewInternInfo because the if conditions are exclusive from each other.

Reply:
Ive been able to debug sPhase now, it seems the issue is from getInterviewInterninfo(). It seems that one of the variables used in getInterviewInternInfo() is null (when we iterate through the response tree, one of the values is null I think it might be when we get the children information), so then getInterviewInternInfo() is trying to run using an empty variable so it cannot produce an output. Could you please review the logic in the method in relation to the responseTree if iterating the way it is currently would get the information from the tree?